src/services/pulls/usePulls.tsx

```diff
@@ -0,0 +1,282 @@
+                  }
+                  changeCoverage
+                }
+                ... on MissingBaseCommit {
+                  message
+                }
+                ... on MissingHeadCommit {
+                  message
+                }
+                ... on MissingComparison {
+                  message
+                }
+                ... on MissingBaseReport {
+                  message
+                }
+                ... on MissingHeadReport {
+                  message
+                }
+              }
+            }
+          }
+          pageInfo {
+            hasNextPage
+            endCursor
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+type GetPullsReturn = { pulls: Array<Pull>; pageInfo: PageInfo | null }
+
+interface UsePullArgs {
+  provider: string
+  owner: string
+  repo: string
+  filters: {
+    state: PullStates
+  }
+  orderingDirection: 'ASC' | 'DESC'
+  opts?: UseInfiniteQueryOptions<GetPullsReturn>
+}
+
+export function usePulls({
+  provider,
+  owner,
+  repo,
+  filters,
+  orderingDirection,
+  opts = {},
+}: UsePullArgs) {
+  const { data, ...rest } = useInfiniteQuery({
+    queryKey: ['pulls', provider, owner, repo, filters, orderingDirection],
+    queryFn: ({ pageParam, signal }) => {
+      return Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: {
+          owner,
+          repo,
+          filters,
+          orderingDirection,
+          after: pageParam,
+        },
+      }).then((res) => {
+        const parsedData = GetPullsSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+          })
+        }
+
+        const data = parsedData.data
+
+        if (data?.owner?.repository?.__typename === 'NotFoundError') {
+          return Promise.reject({
+            status: 404,
+            data: {},
+          })
+        }
+
+        if (data?.owner?.repository?.__typename === 'OwnerNotActivatedError') {
+          return Promise.reject({
+            status: 403,
+            data: {
+              detail: (
+                <p>
+                  Activation is required to view this repo, please{' '}
+                  {/* @ts-expect-error */}
+                  <A to={{ pageName: 'membersTab' }}>click here </A> to activate
+                  your account.
+                </p>
+              ),
+            },
+          })
+        }
+
+        const edges = data?.owner?.repository?.pulls?.edges
+        let branches: Array<Pull | null> = []
+        if (isArray(edges)) {
+          for (const edge of edges) {
+            if (edge?.node) {
+              branches.push(edge.node)
+            }
+
+            if (edge === null) {
+              branches.push(edge)
+            }
+          }
+        }
```

Comment:
I believe you could use `mapEdges` here, although that implementation doesn't check 261 (which makes me wonder why). It must be that connections cannot have null edges in between 🤔 otherwise, we could probably adjust the mapEdges fn or use a utility to mapEdges that accounts for null edges 🤷